packages/core/components/DropZone/index.tsx

```diff
@@ -129,6 +136,22 @@ function DropZoneEdit({ zone, style }: DropZoneProps) {
     }
   }
 
+  if (
+    draggedItem?.draggableId &&
+    allow &&
+    !allow.includes(draggedItem.draggableId)
```

Comment:
`draggableId` is the component ID (i.e. `Heading-1245tgf31f1`), not the component type (`Heading`)

We'll need to use the `getItem()` lib to lookup the item data for the dragged item, and then get the type off there. Suggest we might want to do this outside of the DropZone render lifecycle (via hook or further up tree).

Reply:
Yes, exactly. Further up the tree because the `getItem()` lib is a bit expensive and we might want it elsewhere. We could add do the lookup in <Puck> and add it to our (ever-growing) DropZone context.